dCommon/NiPoint3.h

```diff
@@ -50,111 +62,169 @@ class NiPoint3 {
+		this->x = x;
+	}
 
 	//! Gets the Y coordinate
 	/*!
 	  \return The y coordinate
 	 */
-	float GetY(void) const;
+	constexpr float GetY() const noexcept {
+		return this->y;
+	}
 
 	//! Sets the Y coordinate
 	/*!
 	  \param y The y coordinate
 	 */
-	void SetY(float y);
+	constexpr void SetY(const float y) noexcept {
+		this->y = y;
+	}
 
 	//! Gets the Z coordinate
 	/*!
 	  \return The z coordinate
 	 */
-	float GetZ(void) const;
+	constexpr float GetZ() const noexcept {
+		return this->z;
+	}
 
 	//! Sets the Z coordinate
 	/*!
 	  \param z The z coordinate
 	 */
-	void SetZ(float z);
-
+	constexpr void SetZ(const float z) noexcept {
+		this->z = z;
+	}
 
 	// MARK: Member Functions
 
 	//! Gets the length of the vector
 	/*!
 	  \return The scalar length of the vector
 	 */
-	float Length(void) const;
+	constexpr float Length() const {
+		return std::sqrt(x * x + y * y + z * z);
+	}
 
 	//! Gets the squared length of a vector
 	/*!
 	  \return The squared length of a vector
 	 */
-	float SquaredLength(void) const;
+	constexpr float SquaredLength() const noexcept {
+		return (x * x + y * y + z * z);
+	}
 
 	//! Returns the dot product of the vector dotted with another vector
 	/*!
 	  \param vec The second vector
 	  \return The dot product of the two vectors
 	 */
-	float DotProduct(const Vector3& vec) const;
+	constexpr float DotProduct(const Vector3& vec) const noexcept {
+		return ((this->x * vec.x) + (this->y * vec.y) + (this->z * vec.z));
+	}
 
 	//! Returns the cross product of the vector crossed with another vector
 	/*!
 	  \param vec The second vector
 	  \return The cross product of the two vectors
 	 */
-	Vector3 CrossProduct(const Vector3& vec) const;
+	constexpr Vector3 CrossProduct(const Vector3& vec) const noexcept {
+		return Vector3(((this->y * vec.z) - (this->z * vec.y)),
+			((this->z * vec.x) - (this->x * vec.z)),
+			((this->x * vec.y) - (this->y * vec.x)));
+	}
 
 	//! Unitize the vector
 	/*!
 	  \returns The current vector
 	 */
-	NiPoint3 Unitize(void) const;
+	constexpr NiPoint3 Unitize() const {
+		float length = this->Length();
 
+		return length != 0 ? *this / length : NiPoint3::ZERO;
+	}
 
 	// MARK: Operators
 
 	//! Operator to check for equality
-	bool operator==(const NiPoint3& point) const;
+	constexpr bool operator==(const NiPoint3& point) const noexcept {
+		return point.x == this->x && point.y == this->y && point.z == this->z;
+	}
 
 	//! Operator to check for inequality
-	bool operator!=(const NiPoint3& point) const;
+	constexpr bool operator!=(const NiPoint3& point) const noexcept {
+		return !(*this == point);
+	}
 
 	//! Operator for subscripting
-	float& operator[](int i);
+	constexpr float& operator[](const int i) noexcept {
+		float* base = &x;
+		return base[i];
+	}
 
 	//! Operator for subscripting
-	const float& operator[](int i) const;
+	constexpr const float& operator[](const int i) const noexcept { // TODO: Is this redundant?
+		const float* base = &x;
+		return base[i];
+	}
```

Comment:
yes, it is redundant.  Nothing uses this

Reply:
It is part of the original code though. Maybe there's a reason for it? I'm hesitant to remove it unless there's a pressing need to.

Reply:
This code is just copied I can almost 100% tell you that and ill show you why in discord